### DISCUSSION TITLE
issue with websockets?

### DISCUSSION THREAD

**Discussion by jjmaldonis** [oven-sh/bun#8462] [Q&A]
Hey Bun, I work at [Deepgram](https://deepgram.com/) and while on a recent customer call we discovered that Bun does not support our JS SDK.

You can find reproducible example code here: https://github.com/deepgram/deepgram-js-sdk/blob/main/examples/node-live/index.js
The example accepts microphone audio input and sends the audio to Deepgram. Deepgram then transcribes the audio (speech to text) and returns a JSON message to the user containing the words in the audio.

When running this code via Node you'll see transcripts printed to the terminal.

When running the same code via Bun the websocket connection closes immediately. It looks like Bun has special syntax for websockets. I know the customer tried for days trying to get it to work following your documentation, and he was very tech-savvy.

We wanted to flag this for your team in case it's relevant.

**Comment by hvrsim** [comment]
@jjmaldonis did you ever find a solution for this (or your customer?)

**Comment by nicolas-angelo** [comment]
@jjmaldonis I also tried using Deepgram SDK with bun a few days ago and ran into quite a few issues with websocket connections

**Comment by jjmaldonis** [comment]
@cleanbaja @nicolas-angelo Unfortunately we never found a solution. The customer swapped back to Node and stopped using Bun.

**Comment by yungalgo** [comment]
any follow up here?

**Reply by Bader-Idris** [reply]
this is a good one for deduced [fixes:](https://github.com/oven-sh/bun/issues/28396), to be merged! 🫩
